Bottom: d0761298972631c7ba9b254dd11624038092846c
Top:    068d5afc22029722bb5add0fd40b1448f0b8e651
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-12-19 01:14:52 +0000

action.c: Fix leak of schname in ActionImport()


---

diff --git a/src/action.c b/src/action.c
index 1005b17..69bf89d3 100644
--- a/src/action.c
+++ b/src/action.c
@@ -7706,7 +7706,10 @@ ActionImport (int argc, char **argv, Coord x, Coord y)
       strcat (schname, ".sch");
 
       if (access (schname, F_OK))
-	return hid_action("ImportGUI");
+        {
+          free (schname);
+          return hid_action("ImportGUI");
+        }
 
       sources = (char **) malloc (2 * sizeof (char *));
       sources[0] = schname;
